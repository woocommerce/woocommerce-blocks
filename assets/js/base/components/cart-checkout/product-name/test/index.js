/**
 * External dependencies
 */
import TestRenderer from 'react-test-renderer';

/**
 * Internal dependencies
 */
import ProductName from '..';

describe( 'ProductName', () => {
	test( 'should not render a link if disabled is true', () => {
		const component = TestRenderer.create(
			<ProductName disabled={ true } name={ 'Test product' } />
		);

		expect( component.toJSON() ).toMatchSnapshot();
	} );

	test( 'should render a link if disabled is false', () => {
		const component = TestRenderer.create(
			<ProductName disabled={ false } name={ 'Test product' } />
		);

		expect( component.toJSON() ).toMatchSnapshot();
	} );

	test( 'should render a link if disabled is not defined', () => {
		const component = TestRenderer.create(
			<ProductName name={ 'Test product' } />
		);

		expect( component.toJSON() ).toMatchSnapshot();
	} );
} );
