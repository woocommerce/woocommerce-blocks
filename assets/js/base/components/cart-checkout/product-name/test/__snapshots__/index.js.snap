// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductName should not render a link if disabled is true 1`] = `
<span
  className="wc-block-components-product-name"
>
  Test product
</span>
`;

exports[`ProductName should render a link if disabled is false 1`] = `
<a
  className="wc-block-components-product-name"
>
  Test product
</a>
`;

exports[`ProductName should render a link if disabled is not defined 1`] = `
<a
  className="wc-block-components-product-name"
>
  Test product
</a>
`;
